File: Semester3/Advanced Methods of Programming/Toy Language Interpreter C#/Assignement7/Assignement7/Collection/MyList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Assignement7.Collection
{
    public class MyList<T> : MyIList<T>
    {
        private List<T> lst;

        public MyList(List<T> list)
        {
            this.lst = list;
        }

        public void Add(T element)
        {
            this.lst.Add(element);
        }

        public T Get(int index)
        {
            return this.lst.ElementAt(index);
        }

        public T Remove(int index)
        {
            T el = this.lst.ElementAt(index);
            this.lst.RemoveAt(index);
            return el;
        }

        public bool Remove(T element)
        {
            return this.lst.Remove(element);
        }

        public int Size()
        {
            return this.lst.Count;
        }

        public List<T> ToList()
        {
            return this.lst;
        }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            this.lst.ForEach(e => sb.Append(e.ToString() + "\n"));
            return sb.ToString();
        }
    }
}

File: Semester3/Advanced Methods of Programming/ExamenMAP/src/myCollections/MyList.java
package myCollections;

import java.util.ArrayList;

public class MyList<E> implements MyIList<E> {
    ArrayList<E> list;

    public MyList() {list = new ArrayList<>();}

    @Override
    public int size() {
        return list.size();
    }

    @Override
    public boolean isEmpty() {
        return list.isEmpty();
    }

    @Override
    public boolean add(E element) {
        return list.add(element);
    }

    @Override
    public E get(int index) {
        return list.get(index);
    }

    @Override
    public String toString()
    {

        StringBuilder buffer = new StringBuilder();

        for(E element : list)
            buffer.append(element.toString()).append(" | ");
        return buffer.toString();
    }
}
